istioctl/pkg/multicluster/remote_secret.go

```diff
@@ -334,6 +334,7 @@ func generateServiceAccountYAML(opt RemoteSecretOptions) (string, error) {
 	values := fmt.Sprintf(`
 global:
   istioNamespace: %s
+  externalIstiod: true
```

Comment:
Should this really be default for all flavors of multi-cluster? That doesn't seem right.

Reply:
Actually, I added this by accident in this PR. That said, it's a good question whether it should be added or not. In the external control plane doc, we set `--create-service-account=false` and instead install the service-account first, before we run `create-remote-secret`, which I think is the better way. So whether or not we set it here doesn't matter for that case.

In the general multicluster remote cluster case, however, I'm not sure that it shouldn't be set also. All it does is adds some additional privileges for istiod, esp. the configmap update permission which is needed to allow the istiod to add the ca-root-certificate cm. https://github.com/istio/istio/blob/2d99cd2817b2a939c85806020bd0acc29b261d5e/manifests/charts/istio-control/istio-discovery/templates/reader-clusterrole.yaml#L38

I think this will be needed when the multicluster instructions are changed to create a truly istiodless remote cluster. Currently, a local istiod is being used for the CA.

WDYT, should I remove it?

Reply:
If we want it on for the general case, why have it gated by this value?

Reply:
I think it's not needed for primary-remote MC today, because it's running a local istiod CA on the remote cluster. Once that's changed to a proper istiodless remote (i.e., use the new "external" profile, instead of "remote"), I think the remote istiod may need it. But then it may not be needed at all if certs are configured separately, then istiod won't install the ca-root-cert configmap, right?

Anybody else have more insight on why this is the way it is? The chart has been this way for a quite a while, but then multicluster wasn't working properly with external control plane until recently.

Reply:
I removed the `extenralIstiod: true` setting, since changes to the flag and/or chart is outside the scope of this PR. This PR is only intended to align the default generated sa name with the default istio-remote install.